server/tests/backends/sql_translator/test_sql_translator_steps.py

```diff
@@ -0,0 +1,192 @@
+    return engine
+
+
+def execute(connection, query: str):
+    with connection.cursor() as cursor:
+        cursor.execute(query)
+        # result = cursor.fetchall()
+        df = pd.DataFrame(cursor.fetchall())
+        field_names = [i[0] for i in cursor.description]
+        df.columns = field_names
+        return df
+
+
+def sql_retrieve_city(t):
+    return t
+
+
+def sql_query_describer(t) -> Union[Dict[str, str], None]:
+    lst = t.split(' ')
+    lst = [item for item in lst if len(item) > 0]
+    temp = lst[lst.index('FROM') + 1]
+
+    if len(temp.split('.')) == 2:
+        table_name = temp.split('.')[1]
+    else:
+        table_name = temp.split('.')[0]
+
+    request = f'SELECT column_name as name, data_type as type_code FROM information_schema.columns WHERE table_name = "{table_name}" ORDER BY ordinal_position;'
+    connection = get_connection()
+    with connection.cursor() as cursor:
+        cursor.execute(request)
+        describe_res = cursor.fetchall()
+        res = {r[0]: r[1] for r in describe_res}
+        return res
+
+
+def snowflake_query_describer(t) -> Union[Dict[str, str], None]:
+    connection = get_connection()
+    with connection.cursor() as cursor:
+        describe_res = cursor.describe(t)
+        res = {r.name: type_code_mapping.get(r.type_code) for r in describe_res}
+        return res
+
+
+test_cases = []
+for x in step_cases_files:
+    # Generate a readable id for each test case
+    case_hierarchy = path.dirname(x)[len(fixtures_dir_path) :]
+    case_name = path.splitext(path.basename(x))[0]
+    case_id = case_hierarchy + '_' + case_name
+
+    test_cases.append(pytest.param(case_id, x, id=case_id))
+
+
+# Translation from Pipeline json to SQL query
+@pytest.mark.parametrize('case_id,case_spec_file_path', test_cases)
+def test_sql_translator_pipeline(case_id, case_spec_file_path):
+    spec_file = open(case_spec_file_path, 'r')
+    spec = json.loads(spec_file.read())
+    spec_file.close()
+
+    # TODO - insert data from fixture.json
+    data_to_insert = pd.read_json(json.dumps(spec['input']), orient='table')
+    data_to_insert.to_sql(
+        name=case_id.replace('/', ''),
+        con=get_engine(),
+        index=False,
+        if_exists='replace',
+    )
+
+    steps = spec['step']['pipeline']
+    steps.insert(0, {'name': 'domain', 'domain': f'SELECT * FROM {case_id.replace("/", "")}'})
+    pipeline = Pipeline(steps=steps)
+    pandas_result_expected = pd.read_json(json.dumps(spec['expected']), orient='table')
+    query_expected = spec['other_expected']['sql']['query']
+
+    query, report = translate_pipeline(
+        pipeline,
+        sql_query_retriever=sql_retrieve_city,
+        sql_query_describer=sql_query_describer,  # replace by snowflake_query_describer
+    )
+    result = execute(get_connection(), query)
+    assert query_expected == query
+
+    assert_dataframes_equals(pandas_result_expected, result)
```

Comment:
This function is great!

Maybe adding a few comments to help understanding the different steps:
- inserting data
- translating the query
- executing the query
- comparing result and expected